pkg/ebpftracer/tracer_decode.go

```diff
@@ -99,42 +120,98 @@ func (t *Tracer) decodeAndExportEvent(ctx context.Context, ebpfMsgDecoder *decod
 
 	switch eventId {
 	case events.SchedProcessExec:
-		if err := t.handleSchedProcessExecEvent(&eventCtx, parsedArgs, container, rawEventTime); err != nil {
-			return err
-		}
-	case events.SchedProcessExit, events.ProcessOomKilled:
-		t.handleSchedProcessExitEvent(&eventCtx, container, rawEventTime)
-	case events.SchedProcessFork:
-		t.handleSchedProcessForkEvent(parsedArgs, container)
+		t.handleSchedProcessExecEvent(&eventCtx)
 	default:
 	}
 
 	if _, found := t.signatureEventMap[eventId]; found {
 		t.cfg.SignatureEngine.QueueEvent(event)
 	}
 
-	// Do not process an event we do not have a policy set any further.
-	if _, found := t.eventPoliciesMap[eventId]; !found && t.policy != nil {
-		metrics.AgentSkippedEventsTotal.WithLabelValues(def.name).Inc()
-		return nil
-	}
-
-	if filterPolicy != nil && filterPolicy.filter != nil {
-		if err := filterPolicy.filter(event); err != nil {
-			metrics.AgentSkippedEventsTotal.WithLabelValues(def.name).Inc()
-			return nil
-		}
-	}
-
 	select {
 	case t.eventsChan <- event:
+		metrics.AgentTracerSentEventsTotal.WithLabelValues(def.name).Inc()
 	default:
 		metrics.AgentDroppedEventsTotal.WithLabelValues(def.name).Inc()
 	}
 
 	return nil
 }
 
+var errFoundFingerprint = errors.New("found fingerprint")
+
+func (t *Tracer) getEventArgs(ebpfMsgDecoder *decoder.Decoder, eventCtx types.EventContext) (types.Args, error) {
```

Comment:
Why introduce a completely new way of filtering, when we have policy filters in place already? We will have a LRU cache per container, but this is fine.